src/openforms/registrations/contrib/objects_api/utils.py

```diff
@@ -21,3 +29,43 @@ def html_escape_json(value: T) -> T:
         case _:
             # nothing to do, return unmodified
             return value
+
+
+def check_size_stringyfied_json_not_above_limit(content: str) -> None:
+    if (data_size := sys.getsizeof(content)) > settings.MAX_UNTRUSTED_JSON_PARSE_SIZE:
+        formatted_size = filesizeformat(data_size)
+        max_size = filesizeformat(settings.MAX_UNTRUSTED_JSON_PARSE_SIZE)
+        raise SuspiciousOperation(
+            "Templated out content JSON exceeds the maximum size '%i' (it is %i).",
+            max_size,
+            formatted_size,
+        )
+
+
+def render_to_json(template: str, context: dict[str, Any]) -> JSONValue:
+    """Render the provided template and context into a JSON serializable object"""
+
+    # FIXME: replace with better suited alternative dealing with JSON specifically
+    data = render_from_string(
+        template,
+        context=context,
+        disable_autoescape=True,
+        backend=openforms_backend,
+    )
+
+    check_size_stringyfied_json_not_above_limit(data)
+
+    with catch_json_decode_errors():
+        json_data = json.loads(data)
+
+    return json_data
+
+
+def get_payment_context_data(submission) -> dict[str, Any]:
+    return {
+        "completed": submission.payment_user_has_paid,
+        "amount": str(submission.payments.sum_amount()),
+        "public_order_ids": json.dumps(
+            submission.payments.get_completed_public_order_ids()
+        ),
```

Comment:
ah, that explains...

To be honest, I think this is very confusing for the people making templates. Sometimes a variable already outputs, JSON, other times they need to use a special template tag... It's also hard for us to track what is already serialized as JSON and what isn't.

I'd replace this in the default template with a simple for loop:

```django
[{% for order_id in payment.public_order_ids%}"{{ order_id|escapejs }}"{% if not forloop.last %},{% endif %}{% endfor %}
```